ios/Exponent/Versioned/Core/Api/EXNotifications.m

```diff
@@ -92,41 +93,124 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId kernelServiceDeleg
+      } else {
+        resolve(content.userInfo[@"id"]);
+      }
+    }];
   }];
 
-  resolve(notification.userInfo[@"id"]);
+}
+
+RCT_EXPORT_METHOD(addCategory: (NSString *)categoryId
+                  actions: (NSArray *)actions
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(__unused RCTPromiseRejectBlock)reject)
+{
+  NSMutableArray<UNNotificationAction *> * actionsArray = [[NSMutableArray alloc] init];
+  
+  for(NSArray *action in actions) {
+    int optionsInt = [(NSNumber *)action[2] intValue];
+    UNNotificationActionOptions options = UNNotificationActionOptionForeground + optionsInt;
+                                     
+    if (([action count] == 5)) {
+      UNTextInputNotificationAction * newAction = [UNTextInputNotificationAction actionWithIdentifier:action[0]
+                                                                                                title:action[1]
+                                                                                              options:options
+                                                                                 textInputButtonTitle:action[3] textInputPlaceholder:action[4]];
+      [actionsArray addObject:newAction];
+    } else {
+      UNNotificationAction * newAction = [UNNotificationAction actionWithIdentifier:action[0] title:action[1] options:options];
+      [actionsArray addObject:newAction];
+    }
+  }
+  
+  UNNotificationCategory * newCategory = [UNNotificationCategory categoryWithIdentifier:categoryId actions:actionsArray intentIdentifiers:@[] options:UNNotificationCategoryOptionNone];
+  
+  [[UNUserNotificationCenter currentNotificationCenter] getNotificationCategoriesWithCompletionHandler:^(NSSet * categories) {
+    NSMutableSet * categoriesMutable = [categories mutableCopy];
+    for( UNNotificationCategory * category in categoriesMutable) {
+      if ([category.identifier isEqualToString:categoryId]) {
+        [categoriesMutable removeObject:category];
+        break;
+      }
+    }
+    [categoriesMutable addObject:newCategory];
+    [[UNUserNotificationCenter currentNotificationCenter] setNotificationCategories:categoriesMutable];
+    resolve(@(YES));
+  }];
 }
 
 RCT_EXPORT_METHOD(scheduleLocalNotification:(NSDictionary *)payload
                   withOptions:(NSDictionary *)options
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(__unused RCTPromiseRejectBlock)reject)
 {
-  UILocalNotification *notification = [self _localNotificationFromPayload:payload];
-
-  notification.fireDate = [RCTConvert NSDate:options[@"time"]] ?: [NSDate new];
-  notification.repeatInterval = [RCTConvert NSCalendarUnit:options[@"repeat"]] ?: 0;
-
+  bool repeats = NO;
+  if (options[@"repeats"]) {
+    repeats = ([options[@"repeats"] isEqualToString:@"Yes"])? YES : NO;
```

Comment:
I agree with @sjchmiela we probably want to preserve the existing `repeat` API for now, but if we keep this, the value should be a Boolean instead of the string "Yes"

Reply:
bump, don't compare `options[@"repeats"]` with `"Yes"`, it should be an NSNumber